test/end2end_test.go

```diff
@@ -2671,6 +2671,45 @@ func testExceedMaxStreamsLimit(t *testing.T, e env) {
 	}
 }
 
+const defaultMaxStreamsClient = 100
+
+func TestClientExceedMaxStreamsLimit(t *testing.T) {
+	defer leakCheck(t)()
+	for _, e := range listTestEnv() {
+		testClientExceedMaxStreamsLimit(t, e)
+	}
+}
+
+func testClientExceedMaxStreamsLimit(t *testing.T, e env) {
+	te := newTest(t, e)
+	te.declareLogNoise(
+		"http2Client.notifyError got notified that the client transport was broken",
+		"Conn.resetTransport failed to create client transport",
+		"grpc: the connection is closing",
+	)
+	te.maxStream = 0 // Server allows infinite streams. The cap should be on client side.
```

Comment:
Add something saying that "the server won't send the settings frame for MaxConcurrentStreams"?